src/TSTransformer/nodes/statements/transformEnumDeclaration.ts

```diff
@@ -11,6 +12,21 @@ import { hasMultipleDefinitions } from "TSTransformer/util/hasMultipleDefinition
 import { validateIdentifier } from "TSTransformer/util/validateIdentifier";
 import ts from "typescript";
 
+function needsInverseEntry(member: ts.EnumMember) {
+	if (member.initializer && ts.isStringLiteral(member.initializer)) {
+		const name =
+			ts.isIdentifier(member.name) || ts.isStringLiteral(member.name)
+				? member.name.text
+				: ts.isComputedPropertyName(member.name) && ts.isStringLiteral(member.name.expression)
+				? member.name.expression.text
+				: undefined;
+		if (member.initializer.text === name) {
+			return false;
+		}
+	}
+	return true;
```

Comment:
This code can just be simplified to `typeof checker.getConstantValue(member) !== "string"`

This would also be more correct as this code doesn't handle constant expressions like `1 + 1` or `"A" + "B"`

Reply:
Checking the type of the constant value is not correct - that would mean `Dog = "NotDog"` doesn't get an inverse mapping. Checking the constant value for equality directly is good though.

Reply:
> Especially for enums where some members have inverse lookup allowed by types

This is not the case for any string members

> Removing inverse lookups for strings would be a breaking change,

I don't follow this logic, it's prohibited by the type checker in both roblox-ts and normal TypeScript. It also has zero affect on iteration behavior so any properly written code won't break from removing the unnecessary reverse lookup.

Reply:
> There's nothing wrong with them, and while the types seem picky, it can be done with an `as never`.
Are any and all type assertions not "properly written code"? That seems like a controversial stance on considering it not a breaking change.

Reply:
This is effectively the same as type casting to expose a private member on a class. You can do that, yes, but there's no reason the maintainer should have to prevent your code from breaking by using something that isn't exposed or accessible. I don't consider type assertions properly written code in this case for that reason.

Reply:
That guarantee has always been there in roblox-ts code. I don't think TS behaviour should be leading in our emit. Otherwise, there'd be plenty to reconsider, like `prototype` in classes, not using type-based emit at all, etc etc

Reply:
It's not just TS behavior, it's the type checker prohibiting it which we generally follow. And like I've already said, it is not safe to reverse lookup strings on enums, you *cannot* do a reverse lookup on an enum as there is no guarantee you'll actually do a reverse lookup. 

Using an assertion to do this anyways is already wrong and I don't think there's any harm in removing it to be more consistent with what is actually allowed *by the type checker.*

Reply:
Roblox-ts _has_ provided the guarantee, up to now. There's nothing that needs us removing this, so I don't think we should.

Reply:
Roblox-ts hasn't. You've already encountered the exact issue I'm referring to.

https://discord.com/channels/476080952636997633/494540040991539200/1001144433951113246